core/src/main/scala/com/evolutiongaming/kafka/flow/PartitionFlow.scala

```diff
@@ -206,17 +208,20 @@ object PartitionFlow {
       _ <- Log[F].debug("computing offset to commit")
 
       // find minimum offset if any
-      states <- cache.values
-
-      _ <- Log[F].debug(s"got ${states.size} states from cache")
-
-      stateOffsets <- states.values.toList.traverse { state =>
-        state flatMap (_.context.holding)
-      }
-
-      _ <- Log[F].debug(s"computed stateOffsets: $stateOffsets")
+      pickMin = CommutativeMonoid.instance[Option[Offset]](
```

Comment:
This is a static instance. Let's define it in `private val` somewhere in the object, there's no need to create it every time